common/src/main/java/com/microsoft/identity/common/internal/ui/DualScreenUtil.java

```diff
@@ -0,0 +1,145 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.ui;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.graphics.Rect;
+import android.view.Surface;
+import android.view.WindowManager;
+
+import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.constraintlayout.widget.ConstraintSet;
+
+import com.microsoft.device.display.DisplayMask;
+import com.microsoft.identity.common.R;
+
+import java.util.List;
+
+// Adapt from https://github.com/microsoft/surface-duo-sdk-samples/blob/master/utils/src/main/java/com/microsoft/device/display/samples/utils/ScreenHelper.java
+public class DualScreenUtil {
+
+    // Given a dual screen activity (an activity which uses R.layout.dual_screen_layout),
+    // readjust its child layouts so that it's displayed on both single-screen and dual-screen device correct .
+    public static void adjustLayoutForDualScreenActivity(@NonNull final Activity dualScreenActivity) {
+        final ConstraintLayout dualScreenLayout = dualScreenActivity.findViewById(R.id.dual_screen_layout);
+        if (dualScreenLayout == null) {
+            throw new IllegalArgumentException("This is not a dual screen activity. " +
+                    "Dual screen activity must use R.layout.dual_screen_layout.");
+        }
+
+        int rotation = getRotation(dualScreenActivity);
+        boolean isAppSpanned = isAppSpanned(dualScreenActivity);
+        boolean isHorizontal = rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180;
+
+        final ConstraintSet constraintSet = new ConstraintSet();
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.LEFT, R.id.dual_screen_layout, ConstraintSet.LEFT, 0);
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.RIGHT, R.id.dual_screen_layout, ConstraintSet.RIGHT, 0);
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.TOP, R.id.dual_screen_layout, ConstraintSet.TOP, 0);
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.BOTTOM, R.id.dual_screen_layout, ConstraintSet.BOTTOM, 0);
+
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.LEFT, R.id.dual_screen_layout, ConstraintSet.LEFT, 0);
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.RIGHT, R.id.dual_screen_layout, ConstraintSet.RIGHT, 0);
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.TOP, R.id.dual_screen_layout, ConstraintSet.TOP, 0);
+        constraintSet.connect(R.id.dual_screen_content, ConstraintSet.BOTTOM, R.id.dual_screen_layout, ConstraintSet.BOTTOM, 0);
```

Comment:
these looks exactly the same as the above 4 statements, is this intended?

Reply:
nice catch. 😆